### PR TITLE
.github: check connectivity of external links via GitHub action

### PR DESCRIPTION
**Pull request by yikeke**
### What is changed, added or deleted? (Required)

A migration from https://github.com/pingcap/docs-dm/pull/353

**Note** this external link check isn't required to pass, in order to prevent false alarms from blocking PR merge.

The current regex exclusions is: 

`http://172.*|https://github.com/.*/pull/.*|https://github.com/.*/issues/.*|https://github.com/[^.\n]*$|http://127.0.0.1.*|http://localhost.*|http://.*:.*|https://static.googleusercontent.com/media/research.google.com/.*|.*.md`

Meaning **the check will skip checking the URLs matching the above patterns**.

### Which TiDB version(s) do your changes apply to? (Required)

- [ ] master (the latest development version)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/4464
- Other reference link(s): https://github.com/pingcap/docs/pull/529

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

**Comment by yikeke** [comment]
hi @CaitinChen , the check reports some invalid URLs again in the docs repo. I need your help with fixing URLs in the `adopters.md`:

image

**Comment by CaitinChen** [comment]
1. Please remove this case study
2. https://pingcap.com/case-studies/2017-05-22-Comparison-between-MySQL-and-TiDB-with-tens-of-millions-of-data-per-day
3. https://www.crunchbase.com/organization/g7
4. https://www.crunchbase.com/organization/keruyun-technology-beijing-co-ltd
5. http://treefinancegroup.com.au/
6. https://nn.yjyz.com/

File: adopters.md [hunk 1]
```diff
@@ -72,15 +72,14 @@ This is a list of TiDB adopters in various industries.
...
+|[Tree Finance](http://treefinancegroup.com.au/)|FinTech||
 |[Mashang Consumer Finance](https://www.crunchbase.com/organization/ms-finance)|FinTech||
 |[Snowball Finance](https://www.crunchbase.com/organization/snowball-finance)|FinTech||
-|[Shuangchuang Huipu](http://scphjt.com/)|FinTech||
 |[QuantGroup](https://www.crunchbase.com/organization/quantgroup)|FinTech||
 |[FINUP](https://www.crunchbase.com/organization/finup)|FinTech||
 |[Meili Finance](https://www.crunchbase.com/organization/meili-jinrong)|FinTech||
 |[Guolian Securities](https://www.crunchbase.com/organization/guolian-securities)|Financial Services||
-|[Founder Securities](https://www.linkedin.com/company/founder-securities-co-ltd-/)|Financial Services||
+|[Founder Securities](https://www.crunchbase.com/organization/keruyun-technology-beijing-co-ltd)|Financial Services||
```
Comment (yikeke) [inline adopters.md:82]: PTAL @CaitinChen

**Review by CaitinChen — APPROVED** [review]
LGTM

**Comment by CaitinChen** [comment]
@WT-Liu FYI.

**Pull request merged**